interchain-token-service/src/error.rs

```diff
@@ -0,0 +1,10 @@
+use axelar_wasm_std_derive::IntoContractError;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, IntoContractError)]
+pub enum Error {
+    #[error("failed to decode ITS message: {0}")]
+    InvalidMessage(String),
```

Comment:
I'm not a fan of how the error conversions are handled. This is exactly what error-stack is for:
define it as 
```
#[error("failed to decode ITS message")]
```

and then use
```
result.change_context(Error::InvalidMessage)
```
instead of 
```
result.map_err(|e| Report::new(Error::InvalidMessage(e.to_string())));
```